src/scoping/langs/rust.rs

```diff
@@ -350,7 +352,17 @@ impl FromStr for CustomRustQuery {
     type Err = QueryError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match TSQuery::new(&Rust::lang(), s) {
+        let path = Path::new(s);
+
+        let s: Cow<'_, str> = if path.is_file() {
+            fs::read_to_string(path)
+                .expect("How should QueryError be modified?")
```

Comment:
`QueryError` is part of tree-sitter, and cannot be modified here. I also don't think it should: `from_str` shouldn't be doing I/O. I'm trying to push I/O out as far as possible, aka keeping it out of `lib.rs` and children and contain it in `main.rs`. There, suitable error types (should) already exist, or could be added easily.

Reply:
https://github.com/tree-sitter/tree-sitter/blob/8500e331ebfd49e66dd935b8a9c7a58aba68af37/lib/binding_rust/lib.rs#L256

Reply:
> `QueryError` is part of tree-sitter, and cannot be modified here.

Ah I didn't notice I had drifted into `tree-sitter` when I was looking at `QueryError`.

>  I also don't think it should: `from_str` shouldn't be doing I/O. I'm trying to push I/O out as far as possible, aka keeping it out of `lib.rs` and children and contain it in `main.rs`. There, suitable error types (should) already exist, or could be added easily.

I agree that `FromStr` and `From` impls should not be performing side-effects such as I/O but the current bounds on `CodeQuery` are make it quite difficult / impossible to move the I/O outside of this pipeline.

I've loosened up the pipeline in https://github.com/alexpovel/srgn/pull/144/commits/993c59dd083a550b257b795a6ccec8c1fbd4b4bd.

This involves making a `cli::CodeQuery` and a related `lang::CodeQuery` to keep a similar level of type safety but allow the pipeline to be intercepted in `main.rs` so I can try load a query file based on the path potentially contained in `cli::CodeQuery`. 

See what you think